contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -190,6 +203,12 @@ varbinaryin(PG_FUNCTION_ARGS)
 		PG_RETURN_BYTEA_P(result);
 	}
 
+	if ((!(*pltsql_protocol_plugin_ptr && (*pltsql_protocol_plugin_ptr)->is_tds_client)) && 
+		(len >= 4 && inputText[0] == '\\')) /* COPY command from DMS, use base byteain() directly */
+	{
+		PG_RETURN_BYTEA_P(DirectFunctionCall1(byteain, (Datum) inputText));
```

Comment:
After this change, is the data stored/retrieved during COPY from PSQL and insert from TDS have no difference? Did we validate this?

Reply:
I have manually validated that the data is identical between COPY and a normal TDS insert of the equivalent data, e.g.
<TDS> INSERT INTO t1 VALUES (51)
<PG> COPY dbo.t1 FROM STDIN CSV; -- insert \000\000\000\000\000\000\000\000\000\063
both end up with the same data in the table:
data                   
----------------------
0x00000000000000000033

Reply:
Just checked, and there is one behavior change with this - from the PG endpoint, inserting into a table with an octal string will now use the BYTEA conversion rather than the VARBINARY conversion - however, this can be worked around by directly inserting the VARBINARY value. See below:
```
-- old behavior
babelfish_db=> insert into dbo.t1 values ('\000\063');
INSERT 0 1
babelfish_db=> select * from dbo.t1;
        data
--------------------
 0x5c3030305c303633
(1 row)

-- new behavior
jdbc_testdb=# insert into dbo.t1 values ('\000\063');
INSERT 0 1
jdbc_testdb=# insert into dbo.t1 values ('\x5c3030305c303633');
INSERT 0 1
jdbc_testdb=# select * from dbo.t1;
        data        
--------------------
 0x0033
 0x5c3030305c303633
(2 rows)
```

Ultimately, I don't think this is too much of an issue since this only affects inserting data from the PG endpoint and has a workaround, but it may need to be documented.

Reply:
The workaround you mentioned doesnt look feasible and not at all user friendly

Reply:
The binary value can be derived by doing an ASCII to hex conversion - for example, they could use https://www.rapidtables.com/convert/number/ascii-to-hex.html to get the hex value to insert.